src/dll/ScriptCompiler.cpp

```diff
@@ -99,36 +99,39 @@ const std::string& ScriptCompilerFailure::GetMessage() const
     return m_message;
 }
 
-ScriptCompilerSettings::ScriptCompilerSettings(SccApi& m_scc, std::u8string r6Path)
-    : m_scc(m_scc)
-    , m_r6Path(r6Path)
+ScriptCompilerSettings::ScriptCompilerSettings(SccApi& aApi, std::filesystem::path aR6Path)
+    : m_scc(aApi)
+    , m_r6Path(aR6Path)
 {
 }
 
-ScriptCompilerSettings* ScriptCompilerSettings::AddScriptPath(std::u8string path)
+ScriptCompilerSettings* ScriptCompilerSettings::AddScriptPath(std::filesystem::path aPath)
 {
-    m_scriptPaths.emplace_back(path);
+    m_scriptPaths.emplace_back(aPath);
     return this;
 }
 
-ScriptCompilerSettings* ScriptCompilerSettings::SetCustomCacheFile(std::u8string path)
+ScriptCompilerSettings* ScriptCompilerSettings::SetCustomCacheFile(std::filesystem::path aPath)
 {
-    m_customCacheFile = path;
+    m_customCacheFile = aPath;
     return this;
 }
 
 ScriptCompilerSettings::Result ScriptCompilerSettings::Compile()
 {
-    const auto settings = m_scc.settings_new(reinterpret_cast<const char*>(m_r6Path.c_str()));
+    auto r6PathStr = m_r6Path.u8string();
+    const auto settings = m_scc.settings_new(reinterpret_cast<const char*>(r6PathStr.c_str()));
```

Comment:
Just for my info, what is wrong with:
`const auto settings = m_scc.settings_new(reinterpret_cast<const char*>(r6PathStr.u8string().c_str()));`

I assume `settings_new` will create a copy of the string, right?

Reply:
it does create a copy, but I'm not sure how long temporaries live in C++, so I thought it might be freed before we enter `settings_new`
if this is redundant I can simplify

Reply:
it seems that temporaries live as long as the outermost expression, so what you hinted should work fine, @WSSDude suggested this though, so I just left it unchanged

Reply:
i suggested that when you saved result of c_str to temporary variable and used that after in the call.

that would result in a free, what wopss wrote (and you also had in previous MR if you look at it before review) was fine, it was just one line you for some reason split to two in a wrong way.

Reply:
just to explain myself i guess when i was brought into this lol

Reply:
https://github.com/WopsS/RED4ext/pull/39#discussion_r1431986144

Reply:
oh I see now, I think I just misunderstood your comment